frontends/PyCDE/src/pycde/module.py

```diff
@@ -340,30 +340,48 @@ def __call__(self, op):
     }
 
     # Build the replacement HWModuleOp in the outer module.
-    module_key = str((op.name, sorted(input_ports), sorted(output_ports),
-                      sorted(self.params.items())))
     if "module_name" in self.params:
       module_name = self.params["module_name"]
     else:
-      module_name = op.name
-    if module_key not in self.generated_modules:
+      module_name = self.create_module_name(op, input_ports, output_ports)
+
+    if module_name not in self.generated_modules:
       with mlir.ir.InsertionPoint(mod.regions[0].blocks[0]):
         gen_mod = ModuleDefinition(self.modcls,
                                    module_name,
                                    input_ports=input_ports,
                                    output_ports=output_ports,
                                    body_builder=self.gen_func)
-        self.generated_modules[module_key] = gen_mod
+        self.generated_modules[module_name] = gen_mod
 
     # Build a replacement instance at the op to be replaced.
     with mlir.ir.InsertionPoint(op):
       mapping = {name.value: op.operands[i] for i, name in enumerate(op_names)}
-      inst = self.generated_modules[module_key].create(op.name,
-                                                       **mapping).operation
+      inst = self.generated_modules[module_name].create(op.name,
+                                                        **mapping).operation
       for (name, attr) in attrs.items():
         inst.attributes[name] = attr
       return inst
 
+  def create_module_name(self, op, input_ports, output_ports):
+    op_name = op.name.replace("pycde.", "")
+    input_port_types = "_".join(
```

Comment:
Under the assumption that everything in the module is determined entirely from the parameters, using the input and output ports as the key shouldn't be necessary, yes? I know that's what we were doing before, but I think it was redundant.

Reply:
> Under the assumption that everything in the module is determined entirely from the parameters, using the input and output ports as the key shouldn't be necessary, yes?

Yes, I can remove them. I was adding them to be cautious, but they are probably not necessary in practice. If you define a non-parameterized module with the same name and different port types, that will just mean you're redefining the class at the Python level, which is on you. There is one case this won't handle, which is when you parameterize the module and give it the same name and different port types:

```
@pycde.module
def Parameterized(param):

  class Module:
    x = pycde.Input(pycde.types.i1)
    y = pycde.Output(pycde.types.i1)

@pycde.module
def OtherParams(param):

  class Module:
    x = pycde.Input(pycde.types.i2)
    y = pycde.Output(pycde.types.i2)
```

I think this might not be too common, but it is possible with the current API, and that won't work. I don't think it's worth the extra complexity the types will add to the name unless something like this is happening in practice.

Reply:
Yeah... I agree. I don't think we should allow the above example.